api/envoy/api/v2/auth/cert.proto

```diff
@@ -368,6 +368,9 @@ message UpstreamTlsContext {
   //
   // Defaults to 1, setting this to 0 disables session resumption.
   google.protobuf.UInt32Value max_session_keys = 4;
+
+  // set sni value automatically, operate to set sni field as authority of incoming http connection
+  bool auto_sni = 5;
```

Comment:
I will put this in `Cluster` message or `HttpProtocolOptions`, it affect load balancing due to the connection pooling. It is also hard to get this value before the load balancing selection is done, esp. with transport socket abstraction and matching in cluster.